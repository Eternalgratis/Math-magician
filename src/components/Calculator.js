import React, { useState } from 'react';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [state, setState] = useState(0);
  const clickBtn = (e) => {
    setState((state) => calculate(state, e.target.textContent));
  };

  return (
    <section className="calculator">
      <div className="div-container">
        <div className="screen">
          <span>{state.next || state.total || 0}</span>
        </div>
        <div className="buttons">
          <button className="btn" type="button" onClick={clickBtn}>
            AC
          </button>
          <button className="btn" type="button" onClick={clickBtn}>
            +/-
          </button>
          <button className="btn" type="button" onClick={clickBtn}>
            %
          </button>
          <button className="btn-symbol" type="button" onClick={clickBtn}>
            <span>&#247;</span>
          </button>
          <button className="btn" type="button" onClick={clickBtn}>
            7
          </button>
          <button className="btn" type="button" onClick={clickBtn}>
            8
          </button>
          <button className="btn" type="button" onClick={clickBtn}>
            9
          </button>
          <button className="btn-symbol" type="button" onClick={clickBtn}>
            x
          </button>
          <button className="btn" type="button" onClick={clickBtn}>
            4
          </button>
          <button className="btn" type="button" onClick={clickBtn}>
            5
          </button>
          <button className="btn" type="button" onClick={clickBtn}>
            6
          </button>
          <button className="btn-symbol" type="button" onClick={clickBtn}>
            -
          </button>
          <button className="btn" type="button" onClick={clickBtn}>
            1
          </button>
          <button className="btn" type="button" onClick={clickBtn}>
            2
          </button>
          <button className="btn" type="button" onClick={clickBtn}>
            3
          </button>
          <button className="btn-symbol" type="button" onClick={clickBtn}>
            +
          </button>
          <button className="zero" type="button" onClick={clickBtn}>
            0
          </button>
          <button className="btn" type="button" onClick={clickBtn}>
            .
          </button>
          <button className="btn-symbol" type="button" onClick={clickBtn}>
            =
          </button>
        </div>
      </div>
    </section>
  );
};

export default Calculator;
