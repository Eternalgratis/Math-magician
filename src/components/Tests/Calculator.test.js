import Renderer from 'react-test-renderer';
import Calculator from '../Calculator';

it('Renders Calculator component', () => {
  const tree = Renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});
