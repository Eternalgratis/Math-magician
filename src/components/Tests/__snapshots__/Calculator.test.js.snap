// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Calcu component 1`] = `
<section
  className="calculator"
>
  <div
    className="div-container"
  >
    <div
      className="screen"
    >
      <span>
        0
      </span>
    </div>
    <div
      className="buttons"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="btn-symbol"
        onClick={[Function]}
        type="button"
      >
        <span>
          รท
        </span>
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="btn-symbol"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="btn-symbol"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="btn-symbol"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="btn-symbol"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</section>
`;

exports[`Renders Calculator component 1`] = `
<section
  className="calculator"
>
  <div
    className="div-container"
  >
    <div
      className="screen"
    >
      <span>
        0
      </span>
    </div>
    <div
      className="buttons"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="btn-symbol"
        onClick={[Function]}
        type="button"
      >
        <span>
          รท
        </span>
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="btn-symbol"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="btn-symbol"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="btn-symbol"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="btn-symbol"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</section>
`;

exports[`Renders Home component 1`] = `
<section
  className="calculator"
>
  <div
    className="div-container"
  >
    <div
      className="screen"
    >
      <span>
        0
      </span>
    </div>
    <div
      className="buttons"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="btn-symbol"
        onClick={[Function]}
        type="button"
      >
        <span>
          รท
        </span>
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="btn-symbol"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="btn-symbol"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="btn-symbol"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="btn-symbol"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</section>
`;

exports[`Renders calculator component 1`] = `
<section
  className="calculator"
>
  <div
    className="div-container"
  >
    <div
      className="screen"
    >
      <span>
        0
      </span>
    </div>
    <div
      className="buttons"
    >
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="btn-symbol"
        onClick={[Function]}
        type="button"
      >
        <span>
          รท
        </span>
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="btn-symbol"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="btn-symbol"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="btn-symbol"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="btn"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="btn-symbol"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</section>
`;
